crates/ruma-events/src/beacon.rs

```diff
@@ -0,0 +1,77 @@
+//! Types for extensible live location message events ([MSC3489]).
+//!
+//! [MSC3489]: https://github.com/matrix-org/matrix-spec-proposals/pull/3489
+
+use ruma_common::MilliSecondsSinceUnixEpoch;
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use super::OwnedUserId;
+use crate::location::AssetContent;
+
+/// `BeaconInfoStateEventContent` is a struct that represents the content of a beacon_info state
+/// event. It contains information about a live location sharing event.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "org.matrix.msc3672.beacon", alias = "m.beacon", kind = State, state_key_type = OwnedUserId)]
+pub struct BeaconStateEventContent {
+    /// The description of the location.
+    ///
+    /// It should be used to label the location on a map.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+
+    /// `live` is a boolean that should be true when a user starts sharing location.
+    pub live: bool,
+
+    /// `ts` is an optional `MilliSecondsSinceUnixEpoch` that represents the timestamp of the
+    /// event.
+    #[serde(rename = "org.matrix.msc3488.ts", skip_serializing_if = "Option::is_none")]
+    pub ts: Option<MilliSecondsSinceUnixEpoch>,
+
+    /// `timeout` is a u64 that represents the length of time in milliseconds that the location
+    /// will be live. So the location will stop being shared at `m.ts + timeout` milliseconds
+    /// since the epoch.
+    pub timeout: MilliSecondsSinceUnixEpoch,
```

Comment:
This should be either a `std::time::Duration` (using `#[serde(with = "ruma_common::serde::duration::ms")]`), or `js_int::UInt` since it's not a timestamp which is what `MilliSecondsSinceUnixEpoch` represents.